Source/data/common.hpp

```diff
@@ -0,0 +1,226 @@
+#pragma once

```

Comment:
Let's add a forward iterator interface to simplify file parsing?

```c++
  for (const TsvRecordResult &record : tsv.skipHeader()) {
    if (!record.has_value()) app_fatal(record.errorMessage());
    for (const TsvFieldResult &field : record) {
      if (!field.has_value()) app_fatal(field.errorMessage());
      const std::string_view value = field.str();
      // ...
    }
  }
```

Note that the `operator->` is no longer required for forward iterators, so the implementation can be pretty straightforward.

Reply:
I really struggled to find a good way to declare an iterator-based api that supported the use case I wanted for parsing text files (support columns in arbitrary order, skip columns, don't rescan text unnecessarily. I added a set of iterators for this sort of nested looping returning string_views and some test cases but I don't think it'd be suitable for use in playerdat as is.

Reply:
From what I can tell at a glance, this is already enough for parsing the experience file!
Resolving